Publish minutes of 2023-09-14 meeting (#453)
hober/web-extensions: _minutes/2023-09-12-wecg-tpac.md

 * [devlin] If you install a Chrome and Edge extension the RPID will be based on the origin including the scheme. So even with two extensions with the same ID they will not be compatible.
 * [rew] One proposal is to introduce an equivalent ability to link an extension origin to a domain. The OS for mobile apps validates this link. I'm not sure what would perform this association.
 * [tim cappalli] The browser would be the equivalent to the OS in this model.
 *
 * [tomislav] Is that the main use case? Let's get to what you're trying to achieve.
 * [tim cappalli] We're essentially looking to associate a web domain to an extension-specific scheme.
 * [rew] It could also be used on a web app
 * [tim cappalli] If I create a web application for the Dashlane mobile app, it's automatically associated with the Dashlane origin.
 * [devlin] I discussed this a little with Rew earlier and I wasn't sure why Dashlane doesn't just embed an iframe and use the Dashlane origin from there. That might be viewed as a hack or it might be leveraging the platform.
 * [tim cappalli] That could work, but I would think that is a hack. The user wouldn't see the extension's origin, they'd see the.

>  * [devlin]

Reviewer: Do we have thoughts here or should we remove this line? I don't remember what was said.

Author: I'll leave it in there; it shows that some discussion has taken place, and with the context the discussion can be filed in.

 * [carlos] This would also be useful when importing extensions between browsers (e.g. Firefox has recently introduced a feature to import extensions from Chrome) and there might be more use cases for associations.
 * [devlin] That's interesting. I hadn't considered that. A similar thing we've discussed is not having a host permission confirmation if you can prove you own the domain. For example, if Dashlane can prove they own the extension and domain, then the extension should be able to modify the site because of course it can.
 * [tim cappalli] For clarity, what is a host permission in this context?
 * [devlin] Security boundary in extensions that allows extensions to inject scripts, modify the page, extract content, basically get full access to a set.
 * [tomislav] It can also call navigator.credentials in the context of the web page.
 * [devlin] To me, the existing ability to do so means that it would be misleading if we were to suggest that only one specific extension is in control of the credentials. I don't see any meaningful security benefit from supporting this RP ID in WebAuth.

>  * [??? guy in the back] I'm not too familiar - can the mechanism as described here do that?

Reviewer: (To be consistent with below)

```suggestion
 * [??] I'm not too familiar - can the mechanism as described here do that?
```

 * [nick] (introduces himself as 1Password representative) Yes we can do that.
 * [rew] Maybe I'm missing some of the details, but wouldn't the browser be fetching this data from the domain, not the extension?
 * [devlin] Yes. But me, a malicious extension, can inject whatever I want on a website after the proper authentication has completed. A script can reach in and extract the data it wants from the page.
 * [rew] Doesn't that break the WebAuthN security model?
 * [devlin] Yes
 * [nick] You can't intercept the signing event, but you can intercept the response coming out which is a valid attack.

>  * [carlos] Wouldn't that mean that it's more secure for an extension to perform this work because the extension would get the PR ID rather than a website? Since the PR ID would only be accessible by extensions and linked extensions. While a PR ID from a website could be received by extension content scripts.

Reviewer: ```suggestion
 * [carlos] Wouldn't that mean that it's more secure for an extension to perform this work because the extension would get the RP ID rather than a website? Since the RP ID would only be accessible by extensions and linked extensions. While a RP ID from a website could be received by extension content scripts.
```

 * [nick] If you run malicious code on your own client, we have no meaningful way of interdicting that.
 * [devlin] That's our stance as well. If you run a malicious extension on a site it has access to, well, it has access. To me, this browser validation step is meaningless.
 * [tomislav] It doesn't buy us any security benefit.
 * [devlin] right, this is just an alternative to providing a host permission. Extensions can get exactly the same functionality by just requesting a host permission.
 * [tim c] On the web platform we need both sides. On the extension, we only need the name binding (with the assumption that the security isn't broken)
 * [devlin] I can understand the name binding use case. In that case we take out the browser validation, and bikeshedding aside, allow extensions to specify a RP ID in the manifest.json file, provided that the extension is granted host permissions.
 * [nick] So, the attacker gains a response, they can use it once. Are we assuming continued access?
 * [tim c] Not with a passkey. It's always 0.

>  * &lt;much webauthn terminology>

Reviewer: ```suggestion
* <much webauthn terminology>
```

Author: `<` need to be escaped for Markdown to render correctly. Markdown (at least as rendered on Github) supports HTML, so the best way to ensure that it's rendered correctly is to unconditionally replace unescaped `<` with `&lgt;` (this is done by the tool).

Reviewer: Oh, got it! I thought it was an escaping error in the other direction 😅

>  * [oliver] I don't think it's that hard. As the extension, if you create the private key, you restrict access to it.

Reviewer: ```suggestion
 * [oliver] I don't think it's that hard to get full access to the key. If a malicious extension is present at signup it can use its own key in the `navigation.credentials.create` response.
```

>  * []

Reviewer: ```suggestion
```
